### PR TITLE
Peer to peer fix

### PR DESCRIPTION
**Pull request by enjeyw**
**Describe the Pull Request**

Fix to get peer-to-peer txns working properly.

**Todo**

- [ ] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [ ] Request review from relevant @person or team
- [ ] QA your pull request. This includes running the app, login, testing changes etc.
- [ ] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

File: app/server/me_api/credit_transfer.py [hunk 1]
```diff
@@ -157,6 +158,10 @@ def post(self):
                     }
                     return make_response(jsonify(response_object)), 401
 
+            my_transfer_account = find_transfer_accounts_with_matching_token(
+                g.user, counterparty_transfer_account.token
```
Comment (enjeyw) [inline app/server/me_api/credit_transfer.py:162]: `my_transfer_account_id` isn't always provided by mobile

File: app/server/utils/credit_transfer.py [hunk 2]
```diff
@@ -476,16 +476,13 @@ def check_hash(hash_to_check, transfer_amount, transfer_account_id, user_secret,
 
     intervaled_time = int((unix_time - (unix_time % (time_interval * 1000))) / (time_interval * 1000))
 
-    valid_hash = valid_hmac = False
+    hmac_message = str(transfer_amount) + str(transfer_account_id) + str(intervaled_time)
```
Comment (enjeyw) [inline app/server/utils/credit_transfer.py:479]: hmac formate changed slightly. deprecated hash-based check

**Comment by tristanhcole** [comment]
[Peer to Peer fix](https://trello.com/c/f2jRQldS/579-peer-to-peer-fix)

File: app/server/me_api/credit_transfer.py [hunk 3]
```diff
@@ -229,19 +234,20 @@ def post(self):
...
-            return make_response(jsonify(response_object)), 400
+        if not my_transfer_account:
+            if not my_transfer_account_id:
+                response_object = {
+                    'message': 'You must provide your Transfer Account ID',
+                }
+                return make_response(jsonify(response_object)), 400
 
-        my_transfer_account = TransferAccount.query.get(my_transfer_account_id)
+            my_transfer_account = TransferAccount.query.get(my_transfer_account_id)
```
Comment (michielderoos) [inline app/server/me_api/credit_transfer.py:244]: I might be missing something, but can a user just enter an arbitrary `my_transfer_account_id` here?

**Review by michielderoos — CHANGES_REQUESTED** [review]
Just one note, also should we have a test to cover this?

File: app/server/me_api/credit_transfer.py (hunk 3 above)
Comment (tristanhcole) [reply app/server/me_api/credit_transfer.py:244]: +1

Comment (enjeyw) [reply app/server/me_api/credit_transfer.py:244]: just reviewed. It's definitely a very big problem. I've put in a quick fix and let any clients where we can't confirm whether the bug has been exploited or not know.

Comment (enjeyw) [reply app/server/me_api/credit_transfer.py:244]: We should audit this code properly - it's a bit of a mix of modalities and needs a good refactor i think

File: app/test_app/functional/me_api/test_me_credit_transfer.py [hunk 4]
```diff
@@ -13,3 +15,41 @@ def test_get_me_credit_transfer_api(test_client, create_credit_transfer, create_
                                content_type='application/json', follow_redirects=True)
     assert response.status_code == 201
     assert response.json['data']['credit_transfers'][0]['id'] is create_credit_transfer.id
+
+
+def test_invalid_create_me_credit_transfer_api(
```
Comment (tristanhcole) [inline app/test_app/functional/me_api/test_me_credit_transfer.py:20]: adding a test for a _valid_ credit transfer could be worth it also

**Review by tristanhcole — APPROVED** [review]
Tested API. Don't have phone on me atm.

File: app/test_app/functional/me_api/test_me_credit_transfer.py (hunk 4 above)
Comment (enjeyw) [reply app/test_app/functional/me_api/test_me_credit_transfer.py:20]: that's the one above

**Pull request merged**